queries/futures/useGetFuturesPositionForMarkets.ts

```diff
@@ -57,34 +54,64 @@ const useGetFuturesPositionForMarkets = (options?: UseQueryOptions<FuturesPositi
 			const positionCalls = [];
 			const liquidationCalls = [];
 
+			// isolated margin
 			for (const { market, asset } of futuresMarkets) {
 				positionCalls.push(
 					FMD.positionDetailsForMarketKey(
 						ethersUtils.formatBytes32String(MarketKeyByAsset[asset]),
-						selectedFuturesAddress
+						walletAddress
 					)
 				);
 				const marketContract = new Contract(market, FuturesMarketABI);
-				liquidationCalls.push(marketContract.canLiquidate(selectedFuturesAddress));
+				liquidationCalls.push(marketContract.canLiquidate(walletAddress));
 			}
 
-			const positions = (await ethCallProvider.all(positionCalls)) as PositionDetail[];
-			const canLiquidateState = (await ethCallProvider.all(liquidationCalls)) as boolean[];
-
-			const futuresPositions = [];
-
-			for (let i = 0; i < futuresMarkets.length; i++) {
-				const position = positions[i];
-				const canLiquidate = canLiquidateState[i];
-
-				futuresPositions.push(mapFuturesPosition(position, canLiquidate, assets[i]));
+			// cross margin
+			if (crossMarginAvailable && crossMarginAddress) {
+				for (const { market, asset } of futuresMarkets) {
+					positionCalls.push(
+						FMD.positionDetailsForMarketKey(
+							ethersUtils.formatBytes32String(MarketKeyByAsset[asset]),
+							crossMarginAddress
+						)
+					);
+					const marketContract = new Contract(market, FuturesMarketABI);
+					liquidationCalls.push(marketContract.canLiquidate(crossMarginAddress));
+				}
 			}
 
-			setFuturesPositions(futuresPositions);
+			const positionDetails = (await ethCallProvider.all(positionCalls)) as PositionDetail[];
+			const canLiquidateState = (await ethCallProvider.all(liquidationCalls)) as boolean[];
 
-			return futuresPositions;
+			const isolatedPositions = futuresMarkets
+				.map((futuresMarket: FuturesMarket, ind: number) => {
+					const position = positionDetails[ind];
+					const canLiquidate = canLiquidateState[ind];
+					const asset = assets[ind];
+					return mapFuturesPosition(position, canLiquidate, asset);
+				})
+				.filter(({ remainingMargin }) => remainingMargin.gt(0));
+
+			const crossPositions =
+				crossMarginAvailable && crossMarginAddress
+					? futuresMarkets
+							.map((futuresMarket: FuturesMarket, ind: number) => {
+								const position = positionDetails[futuresMarkets.length + ind];
```

Comment:
This was a bit confusing to follow, I wonder if it might be better to separate isolatedMargin and crossMargin position calls.

Reply:
I think this is a good solution when we move to redux. I opted out of that solution here because they both need to call before the dashboard loads, but if they're running at the same time the can't write to the same object in recoil because of a race condition. 

The other option I considered is to call them separately and store on separate objects in state, then combine using a selector. This avoids the race condition, but makes it so we need to duplicate the query and adds n+1 state variables for the number of futures market types we have.

Reply:
I was thinking more just within this query to separate them in terms of calls, so we could have `isolatedPositionCalls`, `crossPositionCalls`, which would then allow to map from `isolatedPositionDetails` and `crossPositionDetails`, rather than mapping from futuresMarkets both times with offset index. Could then become:

```
const isolatedPositions = isolatedPositionDetails
    .map((position, ind) => {
        const canLiquidate = canLiquidateStateIsolated[ind];
        const asset = assets[ind];
        return mapFuturesPosition(position, canLiquidate, asset);
    })
    .filter(({ remainingMargin }) => remainingMargin.gt(0));

const crossPositions = crossPositionDetails
    .map((position, ind) => {
        const canLiquidate = canLiquidateStateCross[ind];
        const asset = assets[ind];
        return mapFuturesPosition(position, canLiquidate, asset);
    })
    .filter(({ remainingMargin }) => remainingMargin.gt(0));
```

More just a readability thing as it wasn't very clear when I first followed it but definitely not essential.

Reply:
This makes sense. I'll slice the array we get back and rename them, but don't want to split the calls up because it will double our RPC calls.